### PR TITLE
BL-635 Bug-ish/fe-merge-and-optimize-redux-modules

### PR DESCRIPTION
**Pull request by attaris978**
## Description

Complete overhaul of the redux state management system, with removal of deprecated code, correction of minor errors, and modifications to improve intuitive code readability.

1) Updated link to Favicon folder in index.html
2) Commented out unused regional references to optimize mapping functions
3) Combined three separate reducers into one
4) Renamed actions, reducers, components, and myriad other things to make it more intuitive just what they do
5) Renamed folders/files to match what students will be expecting coming out of their Bloomtech education
6) Repaired some missing key attributes for mapped JSX components
7) Removed temporary mounting point for filter components
8) Restructured component files for consistency
9) Killed it. Really. Just slayed the whole thing. Wham. Done. 💥

#### Video Link

https://www.loom.com/share/a28e5bdb38294e96bfcac8f1ff86270e

#### Jira Link

https://bloomtechlabs.atlassian.net/browse/BL-635?atlOrigin=eyJpIjoiN2RmMGNiMjExOTViNGQyZDkyY2E5ZWQ3NjQ0MDAyZjAiLCJwIjoiaiJ9

## Type of change

- [ ✅ ] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ✅ ] My code follows the style guidelines of this project
- [ ✅ ] I have performed a self-review of my own code
- [ ✅ ] I have removed unnecessary comments/console logs from my code
- [ ✅ ] My changes generate no new warnings
- [ ✅ ] I have checked my code and corrected any misspellings
- [ ✅ ] No duplicate code left within changed files
- [ ✅ ] Size of pull request kept to a minimum
- [ ✅ ] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon/favicon.ico" />
```
Comment (Mbravo36) [inline public/index.html:5]: Great use of the favicon.

File: src/components/pages/Table/ContinentSelect.jsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { setContinentFilter } from '../../../state/actions';
+import { setContinentFilter } from '../../../state/actionCreators';
```
Comment (Mbravo36) [inline src/components/pages/Table/ContinentSelect.jsx:3]: The file renaming is very helpful in understanding what exactly it is you're working with.

File: src/state/actionCreators/index.js [hunk 3]
```diff
@@ -1,56 +1,46 @@
-/*Remember separation of concerns.
+/*Consider breaking this down into separate actionCreator files if it becomes overly crowded!
```
Comment (Mbravo36) [inline src/state/actionCreators/index.js:1]: The amount of information passed is very helpful to future tickets and gives a better understanding of how to go about any future information.

**Review by Mbravo36 — APPROVED** [review]
Overall, code was implemented great and looks like a lot of hard work went into putting this all together. Great job y'all.

**Review by MaxwellChrist — APPROVED** [review]
Great work combining the data reducers. Not only does it look better, but the comments you put in the switch statement was helpful. Also the table container file looks much cleaner. All of these changes added up to a great optimization, as well as being summarized flawlessly in the last line of the description section in this pull request

File: src/components/pages/Table/DataTable.jsx [hunk 4]
```diff
@@ -27,10 +41,4 @@ function DataTable({ cases }) {
   );
 }
 
-const mapStateToProps = state => {
```
Comment (willkim76) [inline src/components/pages/Table/DataTable.jsx:30]: I appreciate you standardizing the coding structure of each file. It makes it easier to follow.

File: src/components/pages/Table/SearchBar.jsx [hunk 5]
```diff
@@ -64,15 +74,4 @@ function SearchBar(props) {
   );
 }
 
-const mapStateToProps = state => {
-  const reducerState = state.dataReducer;
-  const filteredReducer = reducerState.filteredReducer;
-
-  return {
```
Comment (willkim76) [inline src/components/pages/Table/SearchBar.jsx:71]: The updated format is much easier to read also the names are more semantic.

File: src/data/filterConstants.js [hunk 6]
```diff
@@ -405,11 +405,11 @@ export const regions = [
     continent: continentEnum.europe,
     geopolitical: geopoliticalEnum.europe,
   },
-  {
```
Comment (willkim76) [inline src/data/filterConstants.js:408]: From the backend perspective, removing all the non-viable regions will get up closer to more manageable query strings.

File: src/state/actionTypes/index.js [hunk 7]
```diff
@@ -1,8 +1,7 @@
-export const GET_DATA = 'GET_DATA';
+//These are for filtering/searching data already retrieved from the API
 export const FILTER_SEARCH = 'FILTER_SEARCH';
-export const RESET_DATA = 'RESET_DATA';
-
-//Backend filter & fetch actions
+export const RESET_CASE_DATA = 'RESET_DATA';
+//Actions to choose filters and fetch API data
```
Comment (willkim76) [inline src/state/actionTypes/index.js:4]: I like how you are providing comments throughout the codebase that provides context for our successors.

**Review by willkim76 — APPROVED** [review]

**Review by MaxwellChrist — APPROVED** [review]
looks like merge conflicts cancelled my last approval. Still stick with what I said before and I approve these changes

File: src/components/pages/Table/GeopoliticalSelect.jsx [hunk 8]
```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { setGeopoliticalFilter } from '../../../state/actions';
+import { setGeopoliticalFilter } from '../../../state/actionCreators';
```
Comment (Ricardorouge) [inline src/components/pages/Table/GeopoliticalSelect.jsx:3]: love the specificity

**Review by Mbravo36 — APPROVED** [review]
Code looks well implemented in new updates.

**Review by Ricardorouge — APPROVED** [review]
thank you for this, having a more organized system will save hundred of hours of headaches when trying to solve merge conflicts

**Pull request merged**